// Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
// This source file is part of the Cangjie project, licensed under Apache-2.0
// with Runtime Library Exception.
//
// See https://cangjie-lang.cn/pages/LICENSE for license information.

// DEPENDENCE:  ../../cjdb_test.py ../../basecase/expr/tryExpr/try_exception.cj
// EXEC: %compiler %cmp_opt %enableCompileDebug %n.cj -o %n
// EXEC: python3 cjdb_test.py %n.info

test#
cjdb,cjdb try_exception,try_exception
b,b 16,try_exception.cj:16
r,r,-> 16
 ,expr try { throw ArithmeticException()  } catch (e: ArithmeticException) { println("an arithmeticException happened: " + e.toString())    } finally { println("the finally block is executed") } ,\(Unit\) \$\d+ = ()
q,q,